src/components/ProgressIndicator/ProgressIndicatorDisplay.js

```diff
@@ -60,7 +60,7 @@ const SidebarStepStatus = ({
   completed = false,
 }) => {
   return (
-    <ProgressItem completed={completed}>
+    <ProgressItem completed={completed} modifiers={isApplicable ? [] : ['not-applicable']}>
```

Comment:
imo it would be a nicer API to have a prop `isApplicable` and let the component `ProgressItem` translate that into the CSS modifiers. In this component, you want to reason in props instead of CSS classes. That also encapsulates the specific CSS class names to the `ProgressItem` component instead of leaking into the `ProgressIndicatorDisplay` component.